data/maps/MP_Scenarios/Island_Hopping.wmf/scripting/hop_island.lua

```diff
@@ -14,9 +14,17 @@ function add_workers(hq, workers)
 end
 
 function add_soldiers(hq, soldiers)
-   local setpoints = {}
+   local setpoints = hq:get_soldiers("all")
    for sdescr, count in pairs(soldiers) do
-      setpoints[ {sdescr:match("(%d):(%d):(%d):(%d)")} ] = count
+      local desc = {sdescr:match("(%d):(%d):(%d):(%d)")}
```

Comment:
Not very nice, but maybe the simplest solution?

```suggestion
      local desc = {}
      for i, v in ipairs({sdescr:match("(%d):(%d):(%d):(%d)")}) do
         desc[i] = tonumber(v)
      end
```

Reply:
found a solution even more simple ;-)

Reply:
Indeed. :)